tf2onnx/tfonnx.py

```diff
@@ -1255,6 +1255,45 @@ def onehot_op(ctx, node, name, args):
         return
 
 
+def onehot_op9(ctx, node, name, args):
+    # T output = OneHot(uint8/int32/int64 input, T depth, T on-value, T off-value, @int axis, @dtype)
+    # tf requires that dtype is same as on-value's and off-value's dtype
+    # in ONNX, op's schema is (input, depth, [off-value, on-value], @int axis)
+    # onnxruntime only supports int64
+    output_dtype = ctx.get_dtype(node.input[2])
+    if output_dtype not in [onnx_pb.TensorProto.INT64, onnx_pb.TensorProto.INT32]:
+        log.warning("unsupported dtype in onnxruntime, onehot can't be used directly")
```

Comment:
we should avoid new logic specifically for onnxruntime. If onnxruntime does not support, we have centralized place to do the cast. e.g. rs6.

Reply:
talked with guohua, I will add check before adding cast node.